prometheus/gc_collector.go

```diff
@@ -0,0 +1,43 @@
+package prometheus
+
+import (
+	"runtime/debug"
+)
+
+type gcCollector struct {
+	numGC      Gauge
+	pauseTotal Gauge
+}
+
+// NewGCCollector returns a collector which exports metrics about the
+// garbage collection stats of the current process.
+func NewGCCollector() *gcCollector {
+	return &gcCollector{
+		numGC: NewGauge(GaugeOpts{
+			Name: "process_gc_num_total",
+			Help: "Total number of garbage collector incovations for the current process.",
```

Comment:
I guess that'd be a single summary called `process_gc_pause_seconds`, which then generates two metrics:

- `process_gc_pause_seconds_count` (the total GC count)
- `process_gc_pause_seconds_sum` (the total number of seconds)

@brian-brazil Not sure though whether I like that more than having two separate explicit counters. From a user perspective, I always felt that explicit (non-summary) counter metrics have the better/clearer names. I know that we're currently using the summary-counter approach quite often, but that's usually in the context of also collecting quantiles via observations. In this case, we don't even have the individual observations/events, only two existing counters that we are bridging. Wouldn't it be better to keep them as normal counters in that case?